src/gui/shell.cpp

```diff
@@ -95,7 +95,7 @@ QString Shell::fontDesc()
 ///
 /// @param updateOption controls update of the guifont option, which should not
 /// be updated when the user calls from 'set guifont=...'.
-bool Shell::setGuiFont(const QString& fdesc, bool force, bool updateOption)
+bool Shell::setGuiFont(const QString& fdesc, bool force, bool updateOption, bool wide)
```

Comment:
I think `guifontwide` should have a separate function `Shell::setGuiFontWide`. This function contains `guifont/GuiFont` specific behavior that you likely don't need to run for `guifontwide`.

I think the only part you need is the `else` block that parses description. We should probably move its own function. This would be a little tricky... Let me know if I can help with this refactor.

Reply:
Of course. I'm glad you can help with this refactor. Actually. I'm not familiar with the behavior of list fonts to fontwide option.

Reply:
Excellent, thanks!

I will put something together that makes it easy for you to write this without `setGuiFont`. I can get you a patch file, or I can commit the changes and you can re-base on top. Let me know if you have a preference.

The code that you need in `setGuiFont` handles the options parsing (`:b`, `:hNN`, etc). We can add a function `GetQFontFromDescription(const QString& description)` and modify the ShellWidget to accept a QFont for `setShellFont(...)` instead of the existing API. You can then write a `Shell::setGuiFontWide` function that handles wide font specific scenarios only.

For the `guifontwide` list behavior (rough code):
```C++
const QStringList wideFontList{ fdesc.split(',') };

for (const auto& fontDescription : wideFontList)
{
    QFont wideFont{ GetQFontFromDescription(fontDescription) };
    // There should probably be some validation here?
    // Also, we might want to update the list only after validating all provided fonts are good?
    m_wideFontList.emplace_back(std::move(wideFont));
}
```

The user should be able to provide multiple wide fonts. Rendering should then pick the first font from that list that contains the double-wide character.

`set guifont=Some\ Font:h11,Another\ Font:h12:b`

We should check `gVim` to make sure our behavior is the same... Ideally, this feature should behave identically.

Reply:
Thank you and I'd like get the helps with your commit.
With your explanation of guifontwide list behavior, I try to add the funtion `Shell::setGuiFontWide`. There's is one point confused: use which pointsize of font on ShellWidget (font's or widefont's)? Currently, `ShellWidget`'s font set to `m_font` by default, and m_fontWide is used only for double-width char as the code below, which may cause the issue that the command `GuiFontWide` won't update the rendering of chars immediately when it's executed.

```cpp
void ShellWidget::setFont(const QFont& f, bool wide)
{
	if (wide)
		this->m_fontWide = f;
	else {
		this->m_font = f;
		QWidget::setFont(f);
	}
}
```

Reply:
What's more, I simply use the result of `ShellWidget::setShellFont` as the validation of wide font. And I think some commit more is required to check wide font containing the double-wide character or not.